### PR TITLE
[stdlib] Changes to support Foundation on OpenBSD.

### PR DESCRIPTION
**Pull request by 3405691582**
* Not implementing POSIXError for a given platform is not a blocking
  problem to getting a successful build of Swift. However, it is
  part of the validation tests (albeit locked behind REQUIRES) and is
  referenced when building Foundation. Implement by forklifting from
  `sys/errno.h`.

* Some Foundation class implementations require some missing includes in
  the platform modulemap. Add these.

### CODE REVIEW

File: stdlib/public/Platform/POSIXError.swift [hunk 1]
```diff
@@ -682,4 +682,202 @@ public enum POSIXErrorCode : Int32 {
...
+    /// Operation not permitted
+    case EPERM			= 1
+    /// No such file or directory
+    case ENOENT			= 2
+    /// No such process
+    case ESRCH			= 3
+    /// Interrupted system call
+    case EINTR			= 4
+    /// Input/output error
+    case EIO			= 5
```
Comment (buttaface) [inline stdlib/public/Platform/POSIXError.swift:698]: What's with the inconsistent formatting?

**Review by finagolfin — APPROVED** [review]

File: stdlib/public/Platform/POSIXError.swift (hunk 1 above)
Comment (3405691582) [reply stdlib/public/Platform/POSIXError.swift:698]: Tabs align the equal signs but they don't seem to be rendering here. Should I use spaces?

Comment (buttaface) [reply stdlib/public/Platform/POSIXError.swift:698]: Don't care.

**Comment by 3405691582** [comment]
@compnerd do you mind reviewing and merging if you approve, or in the alternative, tagging someone who would be best suited to do so?

**Comment by 3405691582** [comment]
ping.

**Comment by 3405691582** [comment]
ping

**Comment by buttaface** [comment]
@spevans, would you review?

**Comment by spevans** [comment]
@swift-ci test

**Comment by 3405691582** [comment]
ping

**Comment by buttaface** [comment]
I think there might be a crunch for the upcoming Swift 5.5 and iPhone 13 releases, so we may have to wait a bit.

**Review by compnerd — APPROVED** [review]
Not sure if it really makes sense to have `mount` as a separate module, but the module definition doesn't need to be cleaned up as part of this change.

**Comment by compnerd** [comment]
@swift-ci please smoke test and merge

**Comment by 3405691582** [comment]
Yeah, we'll look at moving towards the single-header module outside this pr.

**Pull request merged**